File: src/test/java/leetcode_challenges/april/day30/LCCheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTreeTest.java
package leetcode_challenges.april.day30;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class LCCheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTreeTest {
    @Test
    public void test(){
        TreeNode root = new TreeNode(0);
        TreeNode roo9 = new TreeNode(1);
        root.left = roo9;
        TreeNode roo20 = new TreeNode(0);
        root.right = roo20;

        TreeNode root2 = new TreeNode(0);
        roo9.left = root2;

        TreeNode root3 = new TreeNode(1);
        roo9.right = root3;

        LCCheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree c =
                new LCCheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree();

        assertTrue(c.isValidSequence(root,new int[]{0,1,1}));
    }
}

File: src/test/scala/leetcode_challenges/april/day30/LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTreeTest.scala
package leetcode_challenges.april.day30

import org.scalatest.FlatSpec

class LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTreeTest extends FlatSpec{

  "LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree" should
    "check if a String is a valid sequence from root to Leaves Path in a Binary Tree." in {
    val root = new LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.TreeNode(0)
    val roo9 = new LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.TreeNode(1)
    root.left = roo9
    val roo20 = new LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.TreeNode(0)
    root.right = roo20
    val root2 = new LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.TreeNode(0)
    roo9.left = root2
    val root3 = new LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.TreeNode(1)
    roo9.right = root3

    assert(LC_CheckIfaStringIsaValidSequencefromRoottoLeavesPathinaBinaryTree.isValidSequence(root, Array(0,1,1)) === true)
  }
}
